### PR TITLE
Use IC shared lint/prettier config

### PR DESCRIPTION
**Pull request by nathanmarks**
I'm preparing a PR to add typings to the repo and would like to be able to have them linted (and prettified!). Rather than adding the TypeScript eslint plugins and support to the existing config, @dcocchia suggested that converting Snacks to use our shared configs instead of it's own lint config would be a good move (our shared configs already have TS support and prettier integration).

**Most** of the changes here are from prettier/eslint auto fixes but there are some manual changes.

I'll try to point out the manual changes in comments (apart from the `children` prop additions).

**note:** 
- codecov is failing because of the change in # of lines from prettier formatting -- there were no actual changes in functionality or tests
- codeclimate is failing because I fixed a typo in some propTypes and there's repeated code

### CODE REVIEW

File: src/components/Forms/Form.js [hunk 1]
```diff
@@ -43,39 +43,52 @@ class Form extends React.Component {
...
       const component = this.formComponents[name]
       component.setState({
         isValid: false,
-        serverError: serverErrors[name]
+        serverError: serverErrors[name],
       })
     })
   }
 
+  handleSubmit = e => {
```
Comment (nathanmarks) [inline src/components/Forms/Form.js]: manual change: `sort-comp` rule wants `set/get` then `on`/`handle`  after the lifecycle methods

File: src/components/Forms/FormComponent.js [hunk 2]
```diff
@@ -80,11 +86,11 @@ const formComponent = (WrappedComponent) => {
...
       // Check Validator.js validations
       if (isValid && value && validations) {
-        Object.keys(validations).forEach( (validateMethod) => {
+        Object.keys(validations).forEach(validateMethod => {
           const options = validations[validateMethod]
-          const args    = [value].concat(options).filter( v => v ) // Remove null options
+          const args = [value].concat(options).filter(v => v) // Remove null options
 
-          if (!Validator[validateMethod].apply(Validator, args)) {
+          if (!Validator[validateMethod](...args)) {
```
Comment (nathanmarks) [inline src/components/Forms/FormComponent.js]: manual change: `prefer-spread`

File: src/components/Grid/Row.js [hunk 3]
```diff
@@ -9,114 +10,119 @@ const legacyStyles = {
...
     }
   }
 
   return {}
 }
 
 const fullWidthStyles = ['xs', 'sm', 'md', 'mdLg', 'lg', 'xl'].reduce(
-  (styles, size) => ({
-    ...styles,
+  (stylesAcc, size) => ({
```
Comment (nathanmarks) [inline src/components/Grid/Row.js]: manual change: `no-shadow`

File: src/components/Menus/Menu.js [hunk 4]
```diff
@@ -31,37 +31,50 @@ class Menu extends React.Component {
...
+    onSelect: () => {}, // eslint-disable-line no-empty-function
   }
 
   state = {
-    currentTabIndex: null
+    currentTabIndex: null,
   }
 
-  handleBlur = (event) => {
+  getMenuItemChildren() {
```
Comment (nathanmarks) [inline src/components/Menus/Menu.js]: manual change: `sort-comp` rule wants `set/get` then `on`/`handle`  after the lifecycle methods

File: src/components/ScrollTrack/ScrollTrack.js [hunk 5]
```diff
@@ -215,98 +233,116 @@ class ScrollTrack extends Component {
...
-            slideTo: nextForward,
-            parentWidth,
-            trackWidth
-          }
-        }
-      })
+    onBeforeNext(callbackProps).then(response => this.afterBeforeNext(response, callbackProps))
+  }
+
+  afterBeforeNext = (response = {}, callbackProps) => {
```
Comment (nathanmarks) [inline src/components/ScrollTrack/ScrollTrack.js]: manual change: moved this out of `onBeforeNext().then(...)` to fix `no-shadow`

File: src/components/ScrollTrack/ScrollTrack.js [hunk 6]
```diff
@@ -215,98 +233,116 @@ class ScrollTrack extends Component {
...
-            slideTo: nextBack,
-            parentWidth,
-            trackWidth
-          }
-        }
-      })
+    onBeforeBack(callbackProps).then(response => this.afterBeforeBack(response, callbackProps))
+  }
+
+  afterBeforeBack = (response = {}, callbackProps) => {
```
Comment (nathanmarks) [inline src/components/ScrollTrack/ScrollTrack.js]: manual change: moved this out of `onBeforeBack().then(...)` to fix `no-shadow`

File: src/components/Tooltip/Tooltip.js [hunk 7]
```diff
@@ -46,22 +38,25 @@ class Tooltip extends PureComponent {
...
   handleToggle = () => {
-    const {onDismiss, onShow} = this.props
-    this.setState({show: !this.state.show}, () => {
-      if (this.state.show) {
-        onShow()
-      }  else {
-        onDismiss()
+    const { onDismiss, onShow } = this.props
+    this.setState(
+      prevState => ({ show: !prevState.show }),
```
Comment (nathanmarks) [inline src/components/Tooltip/Tooltip.js]: manual change for rule about accessing state in setState

File: src/components/Tooltip/TooltipRootClose.js [hunk 8]
```diff
@@ -40,31 +40,17 @@ class TooltipRootClose extends PureComponent {
...
-
-  removeEventListeners() {
-    if (isNodeEnv()) { return }
-    document.removeEventListener('click', this.handleMouseClick)
-    document.removeEventListener('keyup', this.handleKeyUp)
-    window.removeEventListener('resize', this.handleResize)
-  }
-
-  handleMouseClick = (e) => {
+  handleMouseClick = e => {
```
Comment (nathanmarks) [inline src/components/Tooltip/TooltipRootClose.js]: manual change: `sort-comp` rule wants `set/get` then `on`/`handle`  after the lifecycle methods

File: src/components/Transitions/Fade.js [hunk 9]
```diff
@@ -24,8 +22,7 @@ class Fade extends PureComponent {
      */
     appear: PropTypes.bool,
 
-    /** Axis that is animated */
-    axis: PropTypes.oneOf(['x', 'y']),
```
Comment (nathanmarks) [inline src/components/Transitions/Fade.js]: manual change: unused

File: src/index.js [hunk 10]
```diff
@@ -42,32 +42,28 @@ import Slide from './components/Transitions/Slide'
 import Grow from './components/Transitions/Grow'
 import Fade from './components/Transitions/Fade'
 import Portal from './components/Portal/Portal'
-import Text from './components/typography/Text'
+import Text from './components/Typography/Text'
```
Comment (nathanmarks) [inline src/index.js]: manual change: didn't match filesystem casing

File: src/styles/themer/utils.js [hunk 11]
```diff
@@ -21,42 +21,56 @@ export const defaultTheme = {
...
     secondaryForeground: colors.WHITE,
-    secondaryForegroundFocus: colors.GRAY_46
-  }
+    secondaryForegroundFocus: colors.GRAY_46,
+  },
 }
 
 export function cleanConfig(themeConfig) {
-  const cleanConfig = {}
+  const cleanConfigObj = {}
```
Comment (nathanmarks) [inline src/styles/themer/utils.js]: manual change: renamed for `no-shadow`

File: src/utils/memoize.js [hunk 12]
```diff
@@ -1,11 +1,11 @@
-export default (func) => {
-  const memoizedFunc = function () {
+export default func => {
+  const memoizedFunc = function(...args) {
     let result
-    const key = JSON.stringify(arguments)
+    const key = JSON.stringify(args)
```
Comment (nathanmarks) [inline src/utils/memoize.js]: manual change: lint prefers using `...args` to arguments`

File: src/utils/memoize.js [hunk 13]
```diff
@@ -1,11 +1,11 @@
...
     if (memoizedFunc.cache[key] !== undefined) {
       result = memoizedFunc.cache[key]
     } else {
-      result = func.apply(null, arguments)
+      result = func(...args)
```
Comment (nathanmarks) [inline src/utils/memoize.js]: `prefer-spread`

**Comment by dcocchia** [comment]
Thanks for point out each manual change 🙏🙏 Makes this much nicer to go through :)

File: .eslintrc.js [hunk 14]
```diff
@@ -1,64 +1,36 @@
...
-    "jest": true
+  env: {
+    browser: true,
+    jest: true,
   },
-  "globals": {
-    "__DEV__": true
+  extends: ['@instacart/eslint-config'],
+  rules: {
+    'no-plusplus': OFF,
```
Comment (mmarcuccio) [inline .eslintrc.js]: Is this just to allow for loops? If so, you can use the `allowForLoopAfterthoughts` option.

File: .eslintrc.js [hunk 15]
```diff
@@ -1,64 +1,36 @@
...
+    // Fails on non literal roles attrs. Set to error after this is released:
+    // https://github.com/evcohen/eslint-plugin-jsx-a11y/pull/572
+    'jsx-a11y/no-static-element-interactions': WARN,
+    'react/no-string-refs': OFF,
+    'react/no-find-dom-node': OFF,
```
Comment (mmarcuccio) [inline .eslintrc.js]: For these two rules, should we set to Warn instead? That might prevent future usage. We should be using React.createRef to handle these cases.

Comment (nathanmarks) [reply .eslintrc.js]: We could -- but I made an assumption that `React.createRef` wasn't used to maintain backwards compatibility? (like legacy portal support for eg)

Comment (mmarcuccio) [reply .eslintrc.js]: Good point. We plan on implementing a strategy to allow exposing refs for certain components.  When we do that we will be dropping support for legacy portal as well as upgrading our React peer dependency. We could leave it OFF for now, and when we do that process we could revisit the rule overrides.

**Comment by mmarcuccio** [comment]
@nathanmarks tested locally, and things are working great. 🚢

**Pull request merged**